library/core/src/num/f32.rs

```diff
@@ -1424,9 +1424,19 @@ impl f32 {
     /// ];
     ///
     /// bois.sort_by(|a, b| a.weight.total_cmp(&b.weight));
-    /// # assert!(bois.into_iter().map(|b| b.weight)
-    /// #     .zip([-5.0, 0.1, 10.0, 99.0, f32::INFINITY, f32::NAN].iter())
-    /// #     .all(|(a, b)| a.to_bits() == b.to_bits()))
+    /// ```
+    /// Since the NaN constant does not have a well-defined sign,
+    /// this test may result in either:
+    /// ```
+    /// assert!(bois.into_iter().map(|b| b.weight)
+    ///     .zip([-5.0, 0.1, 10.0, 99.0, f32::INFINITY, f32::NAN].iter())
+    ///     .all(|(a, b)| a.to_bits() == b.to_bits()))
+    /// ```
+    /// or:
+    /// ```
+    /// assert!(bois.into_iter().map(|b| b.weight)
+    ///     .zip([f32::NAN, -5.0, 0.1, 10.0, 99.0, f32::INFINITY].iter())
+    ///     .all(|(a, b)| a.to_bits() == b.to_bits()))
```

Comment:
Instead of separating this out (and getting error messages when it can't be compiled) why not make this logical disjunction part of the code: simply wrap this in

```rust
if f32::NAN.is_sign_negative() {
```

Reply:
Done.